### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by ZsoltBalvanyos**

### CODE REVIEW

File: src/it/scala/composer/StoreIntSpec.scala [hunk 1]
```diff
@@ -81,25 +77,32 @@ class StoreIntSpec extends IntegrationSpec {
...
       .map { client =>
 
-        val responseLogger: Client[IO] => Client[IO] = ResponseLogger.apply0[IO](logBody = true, logHeaders = true)
-        val requestLogger: Client[IO] => Client[IO] = RequestLogger.apply0[IO](logBody = false, logHeaders = true, redactHeadersWhen = _ => false)
+        val responseLogger: Client[IO] => Client[IO] = ResponseLogger[IO](logBody = true, logHeaders = true)
+        val requestLogger: Client[IO] => Client[IO] = RequestLogger[IO](logBody = false, logHeaders = true, redactHeadersWhen = _ => false)
         val loggingClient = responseLogger(requestLogger(client))
 
-        new S3[IO](loggingClient, CredentialsProvider.default[IO], region)
+        S3[IO](loggingClient, CredentialsProvider.default[IO], region)
```
Comment (filosganga) [inline src/it/scala/composer/StoreIntSpec.scala]: The s3 exposes a resource method that creates the client itself and can be easily wrapped into the stream, with `Stream.resource`

File: src/it/scala/composer/rendering/PdfRenderingIntSpec.scala [hunk 2]
```diff
@@ -53,22 +49,25 @@ class PdfRenderingIntSpec extends IntegrationSpec {
...
+        case (client, cfg) =>
+
+          val responseLogger: Client[IO] => Client[IO] = ResponseLogger[IO](logBody = true, logHeaders = true)
+          val requestLogger: Client[IO] => Client[IO] = RequestLogger[IO](logBody = false, logHeaders = true, redactHeadersWhen = _ => false)
 
 
-              PdfRendering[IO](requestLogger(responseLogger(client)), cfg)
-          }
-          .evalMap(f)
+          PdfRendering[IO](requestLogger(responseLogger(client)), cfg)
```
Comment (filosganga) [inline src/it/scala/composer/rendering/PdfRenderingIntSpec.scala]: I am not 100% but the PdfRendering should expose a method that returns a resource and instantiate the HTTP client itself. Not a bid deal anyway, also if you want to add request and response logger you cannot use that method.

File: src/main/scala/com/ovoenergy/comms/composer/Composer.scala [hunk 3]
```diff
@@ -1,156 +1,163 @@
...
-          executionContext = ec,
-          userAgent = Some(
-            `User-Agent`(
-              AgentProduct(s"ovo-energy/comms/${BuildInfo.name}", Some(BuildInfo.version)))),
-          responseHeaderTimeout = 60.seconds,
-        )
-    )
+    BlazeClientBuilder[IO](ec)
+      .withUserAgent(
+        `User-Agent`(AgentProduct(s"ovo-energy/comms/${BuildInfo.name}", Some(BuildInfo.version))))
```
Comment (filosganga) [inline src/main/scala/com/ovoenergy/comms/composer/Composer.scala]: Good stuff!!!

File: src/main/scala/com/ovoenergy/comms/composer/Composer.scala [hunk 4]
```diff
@@ -1,156 +1,163 @@
...
 
-      val buildClient = IO {
-        val builder = AmazonS3Client
-          .builder()
+      endpoint.foreach { uri =>
+        builder
+          .withEndpointConfiguration(new EndpointConfiguration(uri.renderString, region.value))
+          .withPathStyleAccessEnabled(true)
+          // TODO Setup this only if the endpoint is http
+          .withClientConfiguration(new ClientConfiguration().withProtocol(Protocol.HTTP))
```
Comment (filosganga) [inline src/main/scala/com/ovoenergy/comms/composer/Composer.scala]: Please remove the `withPathStyleAccessEnabled` and `withClientConfiguration` they are only useful during the test. Especially the second one: it is exposing the S3 connection to a man-in-the-middle risk.

File: src/main/scala/com/ovoenergy/comms/composer/Composer.scala [hunk 5]
```diff
@@ -1,156 +1,163 @@
...
+      }
+
+      builder.build().asInstanceOf[AmazonS3Client]
```
Comment (filosganga) [inline src/main/scala/com/ovoenergy/comms/composer/Composer.scala]: We need to refactor the templates library to use the comms-aws or/and at least use `AmazonS3`

File: src/main/scala/com/ovoenergy/comms/composer/Composer.scala [hunk 6]
```diff
@@ -1,156 +1,163 @@
...
+      val routes =
+        Router[IO](
+          "/render" -> RenderRestApi[IO](Print.http[IO]).renderService,
+          "/admin" -> AdminRestApi[IO].adminService
+        ).orNotFound
+
+      val http =
+        new BlazeServerBuilder[IO](
+          socketAddress = defaults.SocketAddress,
+          executionContext = ExecutionContext.global,
```
Comment (filosganga) [inline src/main/scala/com/ovoenergy/comms/composer/Composer.scala]: Please use `mainEc` here. It is intended to replace the global in this project as we have some blocking calls for fetch the template.

File: src/main/scala/com/ovoenergy/comms/composer/Composer.scala [hunk 7]
```diff
@@ -1,156 +1,163 @@
...
+          enableWebSockets = true,
+          sslBits = None,
+          isHttp2Enabled = false,
+          maxRequestLineLen = 4 * 1024,
+          maxHeadersLen = 40 * 1024,
+          httpApp = routes,
+          serviceErrorHandler = DefaultServiceErrorHandler[IO],
+          banner = defaults.Banner,
+          channelOptions = ChannelOptions(Vector.empty)
+        ).bindHttp(config.http.port, config.http.host).serve
```
Comment (filosganga) [inline src/main/scala/com/ovoenergy/comms/composer/Composer.scala]: You can override just the value that you need, not all of them.

File: src/main/scala/com/ovoenergy/comms/composer/Composer.scala [hunk 8]
```diff
@@ -1,156 +1,163 @@
...
-                    sms.drain.covaryOutput[StreamApp.ExitCode],
-                    print.drain.covaryOutput[StreamApp.ExitCode]
-                  ).joinUnbounded ++ Stream.emit(StreamApp.ExitCode.Error)
-
-                }
-              }
-          }
-        }
-      }
+    stream.flatten.compile.drain.as(ExitCode.Success)
```
Comment (filosganga) [inline src/main/scala/com/ovoenergy/comms/composer/Composer.scala]: I am not 100% sure if flatten is the best solution here @SystemFw

File: src/main/scala/com/ovoenergy/comms/composer/Config.scala [hunk 9]
```diff
@@ -104,6 +103,11 @@ object Config {
             s"${environment.toStringLowerCase}.aiven.schema_registry.password"),
           credstashF[F, Secret[String]]()(s"${environment.toStringLowerCase}.docraptor.api_key")
         ) { (awsRegion, kafkaSSL, schemaRegistryPassword, docRaptorApiKey) =>
+          val test = environment match {
```
Comment (filosganga) [inline src/main/scala/com/ovoenergy/comms/composer/Config.scala]: What is used for? Maybe you would like to replace the `isTest = environment == Uat` in the docRaptor client?

File: src/main/scala/com/ovoenergy/comms/composer/Config.scala [hunk 10]
```diff
@@ -217,30 +196,13 @@ object Config {
...
-              val consumer = ConsumerSettings(
-                pollTimeout = 500.milliseconds,
-                maxParallelism = Int.MaxValue,
-                nativeSettings = Map(
-                  ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG -> kafkaBootstrapServers,
-                  ConsumerConfig.GROUP_ID_CONFIG -> "comms-composer",
-                  ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG -> "false",
-                  ConsumerConfig.AUTO_OFFSET_RESET_CONFIG -> "earliest"
-                )
+              Kafka.Config(
```
Comment (filosganga) [inline src/main/scala/com/ovoenergy/comms/composer/Config.scala]: Is it the `ConsumerConfig.AUTO_OFFSET_RESET_CONFIG -> "earliest"` retained?

File: src/main/scala/com/ovoenergy/comms/composer/Store.scala [hunk 11]
```diff
@@ -40,14 +37,13 @@ object Store {
     }
   }
 
-  def stream[F[_]: Effect](config: Config): Stream[F, Store[F]] = {
-    Http1Client
-      .stream[F]()
+  def stream[F[_]: ConcurrentEffect](config: Config): Stream[F, Store[F]] = {
+    BlazeClientBuilder(???).stream
```
Comment (filosganga) [inline src/main/scala/com/ovoenergy/comms/composer/Store.scala]: You need to add an `ExecutionContext` as far as I remember to use `BlazeClientBuilder `

File: src/main/scala/com/ovoenergy/comms/composer/http/AdminRestApi.scala [hunk 12]
```diff
@@ -1,28 +1,28 @@
...
+import org.http4s.{BuildInfo => _, _}
 import org.http4s.dsl.Http4sDsl
 import org.http4s.circe._
 import io.circe.literal._
 
-class AdminRestApi[F[_]: Monad] extends Http4sDsl[F] {
+class AdminRestApi[F[_]: Sync] extends Http4sDsl[F] {
 
-  def adminService: HttpService[F] = HttpService[F] {
+  def adminService = HttpRoutes.of[F] {
```
Comment (filosganga) [inline src/main/scala/com/ovoenergy/comms/composer/http/AdminRestApi.scala]: It is better to expose explicit type on public members

File: src/main/scala/com/ovoenergy/comms/composer/http/AdminRestApi.scala [hunk 13]
```diff
@@ -1,28 +1,28 @@
...
     case GET -> Root / "ping" =>
       Ok()
     case GET -> Root / "health" =>
-      Ok(json"""{"status": "healthy"}""")
+      Ok(s"""{"status": "healthy"}""")
```
Comment (filosganga) [inline src/main/scala/com/ovoenergy/comms/composer/http/AdminRestApi.scala]: This will cause the content type to be `text/plain` rather than JSON, why did you change that?

File: src/main/scala/com/ovoenergy/comms/composer/http/RenderRestApi.scala [hunk 14]
```diff
@@ -117,7 +117,7 @@ object RenderRestApi {
 
 class RenderRestApi[F[_]: Sync](render: Render[F]) extends Http4sDsl[F] {
 
-  def renderService: HttpService[F] = HttpService[F] {
+  def renderService = HttpRoutes.of[F] {
```
Comment (filosganga) [inline src/main/scala/com/ovoenergy/comms/composer/http/RenderRestApi.scala]: It is better to expose explicit type on public members

File: src/main/scala/com/ovoenergy/comms/composer/kafka/Kafka.scala [hunk 15]
```diff
@@ -0,0 +1,230 @@
...
+            channel = channel,
+            description = s"Error: ${e.errorCode} ${e.reason}".some,
+            email = none,
+            templateManifest = Some(a.metadata.templateManifest)
+          )
+        }
+      }
+
+      val consumerSettings = (executionContext: ExecutionContext) =>
+        ConsumerSettings(
```
Comment (filosganga) [inline src/main/scala/com/ovoenergy/comms/composer/kafka/Kafka.scala]: It is missing the offset reset to `earliest`

File: src/main/scala/com/ovoenergy/comms/composer/rendering/PdfRendering.scala [hunk 16]
```diff
@@ -85,9 +90,23 @@ object PdfRendering extends Logging {
...
+        val bodyStream: Stream[F, Byte] = Stream
+          .emit(docRaptorBody.asJson)
+          .flatMap(json => fs2.Stream.emits(json.noSpaces.getBytes(StandardCharsets.UTF_8).toSeq))
+          .covary[F]
+
         for {
-          uri <- F.fromEither(Uri.fromString(s"${docRaptorConfig.url}/docs"))
-          request <- POST(uri, body, Authorization(credentials))
+          docRaptorUri <- F.fromEither(Uri.fromString(s"${docRaptorConfig.url}/docs"))
+          request <- F.pure(
```
Comment (filosganga) [inline src/main/scala/com/ovoenergy/comms/composer/rendering/PdfRendering.scala]: You could simply do:
```
request <- POST(docRaptorBody.asJson, docRaptorUri, Authorization(credentials))
```

File: src/servicetest/resources/logback-servicetest.xml [hunk 17]
```diff
@@ -17,16 +17,28 @@
         </encoder>
     </appender>
 
+    <appender name="CONSOLE" class="ch.qos.logback.core.ConsoleAppender">
```
Comment (filosganga) [inline src/servicetest/resources/logback-servicetest.xml]: Do we want to log to console as well? IMHO, it makes really difficult to read the test results.

File: src/test/scala/com/ovoenergy/comms/composer/http/RenderRestApiSpec.scala [hunk 18]
```diff
@@ -36,106 +36,98 @@ class RenderRestApiSpec
...
-      RenderRequest(
-        Map("Foo" -> TemplateData.fromString("bar"))
-      ).asJson
-    )
-
-    val response = req.flatMap(service.apply).futureValue
-    response.status shouldBe Ok
+    new RenderRestApi[IO](successfulRender).renderService.orNotFound
+      .run(
+        Request(
```
Comment (filosganga) [inline src/test/scala/com/ovoenergy/comms/composer/http/RenderRestApiSpec.scala]: You can simplify the request creation:
```scala
POST(RenderRequest(fields).asJson, Uri.uri("/yolo/1.0/Service/print"))
```

File: src/main/scala/com/ovoenergy/comms/composer/model.scala [hunk 19]
```diff
@@ -84,7 +85,7 @@ object model {
     def strings: Fragment[String] = new Fragment[String] {
       // can't reuse the nio charset in `http4s.Charset`, it's private
       def content(s: String): Stream[Pure, Byte] = Stream.chunk(Chunk.bytes(s.getBytes(nioUtf8)))
-      def contentType: ContentType = ContentType(text).withCharset(utf8)
+      def contentType: ContentType = ContentType(plain).withCharset(utf8)
```
Comment (filosganga) [inline src/main/scala/com/ovoenergy/comms/composer/model.scala]: I think it would be better to use: `text.plain` as it represent more the `text/plain` Mime type

File: src/it/scala/composer/StoreIntSpec.scala (hunk 1 above)
Comment (ZsoltBalvanyos) [reply src/it/scala/composer/StoreIntSpec.scala]: We loose the logging middleware then, won't we?

Comment (filosganga) [reply src/it/scala/composer/StoreIntSpec.scala]: Yes we do,
As it is used for the store, we may have the resource method on the store and configure the loggers there.

Not a big deal, just to reduce repetition

File: src/main/scala/com/ovoenergy/comms/composer/http/AdminRestApi.scala (hunk 13 above)
Comment (ZsoltBalvanyos) [reply src/main/scala/com/ovoenergy/comms/composer/http/AdminRestApi.scala]: using json gives this exception:  Invalid JSON in interpolated string

File: src/main/scala/com/ovoenergy/comms/composer/Composer.scala [hunk 20]
```diff
@@ -1,156 +1,148 @@
...
```
Comment (laurence-bird) [inline src/main/scala/com/ovoenergy/comms/composer/Composer.scala]: Moving this to within the for block will remove the need for the flatten

File: src/main/scala/com/ovoenergy/comms/composer/Composer.scala [hunk 21]
```diff
@@ -1,156 +1,148 @@
...
-            // TODO Setup this only if the endpoint is http
-            .withClientConfiguration(new ClientConfiguration().withProtocol(Protocol.HTTP))
-        }
+    Stream.bracket(buildClient)(c => IO(c.shutdown()))
+  }
 
-        builder.build()
+  override def run(args: List[String]): IO[ExitCode] = {
+
+    val stream: Stream[IO, Stream[IO, Any]] = for {
```
Comment (laurence-bird) [inline src/main/scala/com/ovoenergy/comms/composer/Composer.scala]: We can get rid of this any by moving .as(ExitCode.Success) to line 143 i believe

Comment (laurence-bird) [reply src/main/scala/com/ovoenergy/comms/composer/Composer.scala]: or to each of the kafka streams

File: src/it/scala/composer/StoreIntSpec.scala [hunk 22]
```diff
@@ -81,25 +77,23 @@ class StoreIntSpec extends IntegrationSpec {
...
-      .stream[IO]()
-      .map { client =>
 
-        val responseLogger: Client[IO] => Client[IO] = ResponseLogger.apply0[IO](logBody = true, logHeaders = true)
-        val requestLogger: Client[IO] => Client[IO] = RequestLogger.apply0[IO](logBody = false, logHeaders = true, redactHeadersWhen = _ => false)
-        val loggingClient = responseLogger(requestLogger(client))
+    implicit val ec = ExecutionContext.Implicits.global
+    implicit val contextShift = cats.effect.IO.contextShift(ec)
+    implicit val timer: Timer[IO] = IO.timer(ec)
+    implicit val concurrentEffect = IO.ioConcurrentEffect
```
Comment (SystemFw) [inline src/it/scala/composer/StoreIntSpec.scala]: this last implicit should be unneded

File: src/it/scala/composer/StoreIntSpec.scala [hunk 23]
```diff
@@ -81,25 +77,23 @@ class StoreIntSpec extends IntegrationSpec {
...
 
   def withS3[A](f: S3[IO] => IO[A]): IO[A] = {
-    Http1Client
-      .stream[IO]()
-      .map { client =>
 
-        val responseLogger: Client[IO] => Client[IO] = ResponseLogger.apply0[IO](logBody = true, logHeaders = true)
-        val requestLogger: Client[IO] => Client[IO] = RequestLogger.apply0[IO](logBody = false, logHeaders = true, redactHeadersWhen = _ => false)
-        val loggingClient = responseLogger(requestLogger(client))
+    implicit val ec = ExecutionContext.Implicits.global
```
Comment (SystemFw) [inline src/it/scala/composer/StoreIntSpec.scala]: do you need this?

File: src/it/scala/composer/StoreIntSpec.scala [hunk 24]
```diff
@@ -81,25 +77,23 @@ class StoreIntSpec extends IntegrationSpec {
...
+    implicit val contextShift = cats.effect.IO.contextShift(ec)
+    implicit val timer: Timer[IO] = IO.timer(ec)
+    implicit val concurrentEffect = IO.ioConcurrentEffect
 
-        new S3[IO](loggingClient, CredentialsProvider.default[IO], region)
-      }
+    Stream.resource(S3.resource[IO](CredentialsProvider.default[IO], region))
```
Comment (SystemFw) [inline src/it/scala/composer/StoreIntSpec.scala]: The whole block can be just:
`S3.resource[IO](..).use(f)`

File: src/it/scala/composer/rendering/PdfRenderingIntSpec.scala [hunk 25]
```diff
@@ -53,22 +49,25 @@ class PdfRenderingIntSpec extends IntegrationSpec {
...
-          .zip(Stream.eval(docRaptorConfig))
-          .map {
-            case (client, cfg) =>
 
-              val responseLogger: Client[IO] => Client[IO] = ResponseLogger.apply0[IO](logBody = true, logHeaders = true)
-              val requestLogger: Client[IO] => Client[IO] = RequestLogger.apply0[IO](logBody = false, logHeaders = true, redactHeadersWhen = _ => false)
+    val ec = ExecutionContext.Implicits.global
+    implicit val contextShift = IO.contextShift(ec)
+    implicit val timer: Timer[IO] = IO.timer(ec)
+    implicit val concurrentEffect = IO.ioConcurrentEffect
```
Comment (SystemFw) [inline src/it/scala/composer/rendering/PdfRenderingIntSpec.scala]: no need for the instance

File: src/servicetest/resources/logback-servicetest.xml (hunk 17 above)
Comment (SystemFw) [reply src/servicetest/resources/logback-servicetest.xml]: yeah no logs in test pls
